### PR TITLE
chore(test): Use GITHUB_TOKEN for higher rate limit

### PR DESCRIPTION
**Pull request by lorensr**
## Problem

CI is rate limited by Github REST API:

```
@temporalio/testing: > node ./scripts/download-test-server.mjs
@temporalio/testing: internal/process/esm_loader.js:74
@temporalio/testing:     internalBinding('errors').triggerUncaughtException(
@temporalio/testing:                               ^
@temporalio/testing: HTTPError: Response code 403 (rate limit exceeded)
```
[build-and-test (14, macos-latest)](https://github.com/temporalio/sdk-typescript/runs/6314570053?check_suite_focus=true#logs)

Currently on the default anonymous 60 requests per hour per IP address.

## Solution

Authenticate, to get 1k requests per hour per repo:

https://docs.github.com/en/rest/overview/resources-in-the-rest-api#requests-from-github-actions

### CODE REVIEW

File: packages/testing/scripts/download-test-server.mjs [hunk 1]
```diff
@@ -24,6 +24,20 @@ const defaultHeaders = {
   'User-Agent': '@temporalio/testing installer',
 };
 
+const { GITHUB_TOKEN } = process.env;
+if (GITHUB_TOKEN) {
+  console.log(`Using GITHUB_TOKEN ${GITHUB_TOKEN}`);
```
Comment (bergundy) [inline packages/testing/scripts/download-test-server.mjs:29]: You definitely don't want to log the secret token

**Comment by lorensr** [comment]
Looks like it's working, although with a higher limit, which implies it's either a personal access token or we're an Enterprise Cloud org: 

https://docs.github.com/en/rest/overview/resources-in-the-rest-api#requests-from-personal-accounts

```
@temporalio/testing: Using GITHUB_TOKEN
@temporalio/testing: Rate limit status: 14959 requests remaining
```

[build-and-test (14, ubuntu-latest)](https://github.com/temporalio/sdk-typescript/runs/6315431267?check_suite_focus=true#logs)

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -35,6 +35,7 @@ jobs:
       - run: npm run build
         env:
           BUILD_CORE_RELEASE: true
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
Comment (bergundy) [inline .github/workflows/ci.yml:38]: Could you set this for the entire pipeline? Just put this at the top level

File: packages/testing/scripts/download-test-server.mjs [hunk 3]
```diff
@@ -24,6 +24,20 @@ const defaultHeaders = {
   'User-Agent': '@temporalio/testing installer',
 };
 
+const { GITHUB_TOKEN } = process.env;
+if (GITHUB_TOKEN) {
+  console.log(`Using GITHUB_TOKEN`);
+  defaultHeaders['Authorization'] = `Bearer ${GITHUB_TOKEN}`;
+}
+
+const rateLimitResult = await got('https://api.github.com/rate_limit', {
```
Comment (bergundy) [inline packages/testing/scripts/download-test-server.mjs:33]: Let's not put this here, it's nice for CI but we don't want this for every user installing the package

**Review by bergundy — COMMENTED** [review]
image

**Review by bergundy — APPROVED** [review]

**Pull request merged**